packages/jaeger-ui/src/utils/span-ancestor-ids.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import _filter from 'lodash/filter';
+import _find from 'lodash/find';
+import _get from 'lodash/get';
+import _map from 'lodash/map';
+
+import type { Span } from '../types/trace';
+
+function getFirstAncestor(span: Span): ?Span {
+  const referencesWithASpan = _filter(span.references, 'span.spanID');
+  const firstAncestorSpan = _find(
+    referencesWithASpan,
+    ({ refType }) => refType === 'CHILD_OF' || refType === 'FOLLOWS_FROM'
+  );
+  return _get(firstAncestorSpan, 'span');
+}
```

Comment:
I used lodash functions instead of a loop to avoid needing to check `Array.isArray` myself.  It's not hard to do these checks, but it's easier to delegate that responsibility to lodash and trust that it's done correctly rather than needing to write the same checks over and over.

Reply:
I cleaned up the implementation to a `_find` with a better check. I prefer this as it makes the code more expressive. Seeing a `_find` with a condition it is clear that the function will return the first span in the references array that satisfies the boolean or `undefined` if none do or `undefined` if references is not an array.